src/rmp/src/Restructure.cpp

```diff
@@ -186,104 +187,56 @@ void Restructure::runABC()
-                 "remap",
-                 1,
-                 "Writing ABC script file {}.",
-                 abc_script_file);
-      if (writeAbcScript(abc_script_file)) {
-        std::string abc_command = std::string("yosys-abc -F ") + abc_script_file;
-        if (logfile_ != "")
-          abc_command
-              = abc_command + " >& " + logfile_ + std::to_string(temp_mode_idx);
-
-        pid_t child_pid = fork();
-        if (child_pid == 0) {  // Begin child
-          // Run in child process
-          int ret = execlp("sh", "sh", "-c", abc_command.c_str(), 0);
-          // Execution of command failed
-          logger_->error(
-              RMP,
-              31,
-              "Failed to run ABC with exit code {}. Please check the "
-              "messages for details.",
-              ret);
-          exit(ret);
-        }  // End child
-
-        if (child_pid > 0) {
-          child_proc[temp_mode_idx] = child_pid;
-        } else if (child_pid < 0) {
-          logger_->warn(
-              RMP,
-              29,
-              "Failed to create new ABC process, could not fork parent "
-              "process. Please check OS messages for details.");
-        }
-
-        files_to_remove.emplace_back(abc_script_file);
-      }
-    }  // end spawn
+	     logger_, RMP, "remap", 1, "Running ABC with {} modes.", modes.size());
 
-    // Wait for ABC process(es)
-    for (int curr_thread = 0; curr_thread < max_parallel_runs; ++curr_thread) {
-      int child_idx = curr_mode_idx + curr_thread;
-      pid_t child = child_proc[child_idx];
+  for (size_t curr_mode_idx = 0; curr_mode_idx < modes.size(); curr_mode_idx++) {
+    output_blif_file_name_
+      = work_dir_name_ + std::string(block_->getConstName())
+      + std::to_string(curr_mode_idx) + "_crit_path_out.blif";
 
-      if (child == 0) {
-        continue;
-      }
+    opt_mode_ = modes[curr_mode_idx];
 
-      int return_status;
-      waitpid(child, &return_status, 0);
-
-      if (return_status) {
-        child_proc[child_idx] = 0;
-        logger_->warn(
-            RMP,
-            15,
-            "ABC failed with code {}. Please check {} log file for details.",
-            return_status,
-            logfile_ + std::to_string(child_idx));
-      }
-    }  // end wait
+    std::string abc_script_file = work_dir_name_
+      + std::to_string(curr_mode_idx)
+      + "ord_abc_script.tcl";
+    if (logfile_ == "")
+      logfile_ = work_dir_name_ + "abc.log";
 
-    curr_mode_idx += max_parallel_runs;
+    debugPrint(logger_,
+	       RMP,
+	       "remap",
+	       1,
+	       "Writing ABC script file {}.",
+	       abc_script_file);
+
+    if (writeAbcScript(abc_script_file)) {
+      // call linked abc
+      Abc_Frame_t * abc_frame;
+      Abc_Start();
+      abc_frame = Abc_FrameGetGlobalFrame();
+      std::string command = "source " + abc_script_file;
+      child_proc[curr_mode_idx] = Cmd_CommandExecute( abc_frame, command.c_str() );
```

Comment:
Why do we have to write a file if it is linked (and later read a blif)?  Is that something for a future PR?
Should child_proc be abc_return_code ?  I don't think there is a child proc anymore.

Reply:
Yes, this is a future PR. I wanted to start with this milestone to get rid of the fork and get a unit test in.